crates/biome_css_analyze/src/lint/nursery/use_consistent_grid_areas.rs

```diff
@@ -0,0 +1,256 @@
+                    state.span,
+                    markup! {
+                        "Duplicate filled in rectangle are not allowed"
+                    },
+                )
+                .note(markup! {
+                    "Consider removing the duplicated filled-in rectangle: " <Emphasis>{state.text.as_ref().unwrap()}</Emphasis>
+                }),
+            ),
+        }
+    }
+}
+
+// Check if the grid areas are consistent
+fn is_consistent_grids(grid_areas_props: GridAreasProps) -> Option<UseConsistentGridAreasState> {
+    let first_prop = clean_text(&grid_areas_props[0].0);
+    let first_len = first_prop.len();
+    let mut shortest = &grid_areas_props[0];
+
+    for grid_areas_prop in &grid_areas_props {
+        let cleaned_text = clean_text(&grid_areas_prop.0);
+        // Check if the grid areas are empty
+        if cleaned_text.is_empty() {
+            return Some(UseConsistentGridAreasState {
+                text: None,
+                span: grid_areas_prop.1,
+                reason: GridAreaValidationError::EmptyGridArea,
+            });
+        }
+        // Check if all elements have the same length
+        if cleaned_text.len() != first_len {
+            if cleaned_text.len() < clean_text(&shortest.0).len() {
+                shortest = grid_areas_prop;
+            }
+            return Some(UseConsistentGridAreasState {
+                text: None,
+                span: shortest.1,
+                reason: GridAreaValidationError::InconsistentCellCount,
+            });
+        }
+    }
+
+    // Check if there are no duplicate grid tokens
+    // It should be partial match because for example, in the following grid areas:
+    // {"a a a"
+    //  "b b b"; }
+    //  are the consistent grid properties because it forms a single filled-in rectangle.
+    if grid_areas_props
+        .iter()
+        .all(|prop| is_all_same(prop.0.clone()))
+    {
+        return None;
+    }
+    //  But in the following grid areas:
+    //  {"a a a"
+    //   "b b a"; }
+    //   are not consistent because `a` breaks a single filled-in rectangle.
+    if let Some(result) = has_partial_match(&grid_areas_props) {
+        return Some(UseConsistentGridAreasState {
+            text: Some(result.0),
+            span: result.1,
+            reason: GridAreaValidationError::DuplicateGridToken,
+        });
+    }
+
+    None
+}
+
+// Check if all characters in a string are the same
+fn is_all_same(token_text: TokenText) -> bool {
+    let prop = clean_text(&token_text);
+    let mut chars: Vec<char> = prop.chars().collect();
+
+    // Sort the characters
+    chars.sort_unstable();
+
+    // Check if the first and last characters are the same
+    chars
+        .iter()
+        .filter(|&&c| !c.is_whitespace())
+        .take(1)
+        .chain(chars.iter().rev().filter(|&&c| !c.is_whitespace()).take(1))
+        .collect::<Vec<_>>()
+        .windows(2)
+        .all(|w| w[0] == w[1])
+}
```

Comment:
It's too complicated. How about checking if all the chars are the same as the first char instead?
```rs
let chars = prop.chars().filter(|c| !c.is_whitespace()).collect();
let head = chars[0]
chars.iter().all(|&c| c == head)
```